### PR TITLE
feat: add `regular_character`

### LINKED ISSUE #3107: dualizing does not work for free resolutions
Example due to @wdecker :
```julia
X = veronese();
I = defining_ideal(X);
Pn = base_ring(I)
FI = free_resolution(I)
F = graded_free_module(Pn, 1)
dualFIC = hom(FI.C, F)
```
I suppose the problem is the following: The modules in `FI.C` have different concrete type. Now a list is created from those modules in an incremental way in `src/Modules/UngradedModules/HomologicalAlgebra.jl:361`  with the first element giving a type to that list. The second then does not match anymore.

### PR DESCRIPTION
**Pull request by thofma**
CC: @joschmitt

### CODE REVIEW

File: src/Groups/group_characters.jl [hunk 1]
```diff
@@ -1724,6 +1724,28 @@ function trivial_character(G::GAPGroup)
...
+
+julia> values(regular_character(G))
+3-element Vector{QQAbElem{AbsSimpleNumFieldElem}}:
+ 6
+ 0
+ 0
+```
+"""
+function regular_character(G::GAPGroup)
+  ccl = conjugacy_classes(G)
```
Comment (thofma) [inline src/Groups/group_characters.jl]: @ThomasBreuer is this correct? I saw that the other functions used `conjugacy_classes(character_table(G))`.

**Comment by thofma** [comment]
@ThomasBreuer What do you think of exporting `class_function`?

File: src/Groups/group_characters.jl (hunk 1 above)
Comment (thofma) [reply src/Groups/group_characters.jl]: OK, I read some docstring and understand why I need the character table

**Comment by joschmitt** [comment]
This would resolve #3390 assuming we export `class_function`, right?

**Comment by thofma** [comment]
Exporting `class_function` would be more complicated, so let's keep this PR here focused on adding `regular_character`.

**Review by ThomasBreuer — COMMENTED** [review]
Actually `regular_character` can assume that the class of the identity element is the first one, thus the function need not do computations with the group.
(I should state this fact in the docstring for `conjugacy_classes` for character tables.)

What about providing `regular_character(tbl::GAPGroupCharacterTable)`,
and defining `regular_character(G::GAPGroup) = regular_character(character_table(G))`?

**Comment by fingolfin** [comment]
I think what @ThomasBreuer suggests would be the way to go. Of course we can also do that later if you need this in immediately.

**Comment by thofma** [comment]
Adjusted following your comments. Please have a look again @ThomasBreuer

File: src/Groups/group_characters.jl [hunk 2]
```diff
@@ -1724,6 +1724,48 @@ function trivial_character(G::GAPGroup)
...
+```
+"""
+function regular_character(G::GAPGroup)
+  return regular_chacter(character_table(G))
+end
+
+@doc raw"""
+    regular_character(tbl::GAPGroupCharacterTable)
+
+Return the regular character of `G`.
```
Comment (ThomasBreuer) [inline src/Groups/group_characters.jl]: "of `tbl`"

**Review by ThomasBreuer — APPROVED** [review]
Looks good. Thanks. Just one minor comment on a docstring.

File: src/Groups/group_characters.jl (hunk 2 above)
Comment (thofma) [inline src/Groups/group_characters.jl]:
```suggestion
Return the regular character of `tbl`.
```

File: src/Groups/group_characters.jl [hunk 3]
```diff
@@ -1724,6 +1724,48 @@ function trivial_character(G::GAPGroup)
...
+
+julia> values(regular_character(G))
+3-element Vector{QQAbElem{AbsSimpleNumFieldElem}}:
+ 6
+ 0
+ 0
+```
+"""
+function regular_character(G::GAPGroup)
+  return regular_chacter(character_table(G))
```
Comment (thofma) [inline src/Groups/group_characters.jl]:
```suggestion
  return regular_character(character_table(G))
```

File: docs/src/Groups/group_characters.md [hunk 4]
```diff
@@ -150,6 +151,7 @@ natural_character(G::Union{MatrixGroup{QQFieldElem}, MatrixGroup{AbsSimpleNumFie
 natural_character(G::MatrixGroup{T, MT}) where T <: FinFieldElem where MT
 natural_character(rho::GAPGroupHomomorphism)
 trivial_character(G::GAPGroup)
+regular_character(G::GAPGRoup)
```
Comment (thofma) [inline docs/src/Groups/group_characters.md]:
```suggestion
regular_character(G::GAPGroup)
```

File: src/Groups/group_characters.jl [hunk 5]
```diff
@@ -1724,6 +1724,48 @@ function trivial_character(G::GAPGroup)
...
+julia> values(regular_character(tbl))
+3-element Vector{QQAbElem{AbsSimpleNumFieldElem}}:
+ 6
+ 0
+ 0
+```
+"""
+function regular_character(tbl::GAPGroupCharacterTable)
+  val = ZZRingElem[0 for i in 1:length(tbl)]
+  val[1] = order(G)
```
Comment (thofma) [inline src/Groups/group_characters.jl]:
```suggestion
  val[1] = order(group(tbl))
```

**Review by fingolfin — APPROVED** [review]

File: src/Groups/group_characters.jl [hunk 6]
```diff
@@ -1724,6 +1724,48 @@ function trivial_character(G::GAPGroup)
...
+end
+
+@doc raw"""
+    regular_character(tbl::GAPGroupCharacterTable)
+
+Return the regular character of `tbl`.
+
+# Examples
+```jldoctest
+julia> tbl = character_table(symmetric_group(3))
```
Comment (fingolfin) [inline src/Groups/group_characters.jl]:
```suggestion
julia> tbl = character_table(symmetric_group(3));
```

**Pull request merged**